### ISSUE TITLE
DJI Air 2s IMU Calibration

### ISSUE THREAD

**Issue by umair1988** [o-gs/dji-firmware-tools#296]
Hi all,
I took up a project to fix a crashed DJI Air 2s drone. After having replaced the left rear leg and some repairs on the gimbal the aircraft was up again flying but after a few minutes of flight the Air2s displayed an IMU error during the flight and I immediately landed that aircraft safely. I tried the IMU calibration but it always gets stuck at 6% followed by an IMU calibration error. I tried replacing the GPS module, IMU module and GPS flex cable but no joy. The aircraft currently display errors 30055 (IMU not calibrated. Calibrate IMU) and 30056 (IMU calibration incomplete. Recalibrate IMU). I feel it's a software error and some software glitch is stopping me from getting rid of the error. Can the Github community please look into the issue and help me out, please? Also while doing an extensive google search I came across a screenshot of some software kit which hints at calibrating IMU through some software. The screenshot of errors and the potential software that may allow IMU calibration is enclosed. If someone had faced similar errors in the past and or knows something about the software please help out. Many thanks

![Screenshot_20220915-021910](https://user-images.githubusercontent.com/62441729/191871883-8e959946-32a6-4eb2-8922-2c4b900dfc98.jpg)

![58301472-3b79bb00-7def-11e9-945c-091f854a1f51](https://user-images.githubusercontent.com/62441729/191871767-46e370fc-91e6-425f-974b-53067fcfd857.png)

**Comment by umair1988** [comment]
Any help please 🙏

**Comment by rbeck425** [comment]
These tools on this page (can be downloaded) are what you need. That program above is just a fancy front end to hit buttons. Look on youtube and will find many people using these tools for the Mavic Air 2 with great success. Give it a shot!

**Comment by rbeck425** [comment]
https://www.youtube.com/watch?v=6MMpbMhy20Y

**Comment by umair1988** [comment]
@rbeck425 thank you for the reply. The video is indeed helpful but it's related to Gimbal calibration. While I have gimbal calibration error but my Air 2s gimbal is pointing in the right direction. My issue is related to IMU calibration which fails no matter what. So question that is there a python code for fixing IMU calibration? It seems that the GUI picture I pasted above has some sort of IMU tool for calibration. So @rbeck425 can you please guide me with regards to IMU calibration tool or python code for same please 🙏? Standing by for your valuable input

**Comment by rbeck425** [comment]
Are you saying that the warning message will not go away but the gimbal works correctly after using the python code that is packaged with the DJI-Firmware-Tools?

**Comment by umair1988** [comment]
Yes that is what I am trying to say partially. Gimbal is working and warning error 40011 remains for which I am not bothered as it doesn't effect drone operations. My concern is error 30055 & 30056 which are related to IMU module connected to GPS chip. My drone fails to calibrate IMU and error 30055 & 30056 are stopping the drone from taking a flight. Is there any code for calibrating the IMU module?

**Comment by umair1988** [comment]
Can someone please guide with IMU calibration error 30055 & 30055 🙏

**Comment by alxgacc** [comment]
Who needs remove error 40011 or 40021, you can write me on WhatsApp +62 812-8430-9143

**Comment by PastorNate** [comment]
I am with you on this issue and have been searching high and low for this.  Any luck for you?

**Comment by SandieJacob** [comment]
To remove 40011 error. Please contact whatsapp +919074728363

**Comment by Th3SnowyOwl** [comment]
> ![58301472-3b79bb00-7def-11e9-945c-091f854a1f51](https://user-images.githubusercontent.com/62441729/191871767-46e370fc-91e6-425f-974b-53067fcfd857.png)
Please tell me where you found this software!

**Comment by blanqui1801** [comment]
Amigo si pudo dar solución al error de la imu.. con código 30055 y 30056

**Comment by blanqui1801** [comment]
> Hola a todos, me embarqué en un proyecto para reparar un dron DJI Air 2s averiado. Tras sustituir la pata trasera izquierda y reparar el estabilizador, la aeronave volvió a despegar, pero tras unos minutos de vuelo, el Air 2s mostró un error de IMU y aterricé inmediatamente sin problemas. Intenté calibrar la IMU, pero siempre se queda atascada al 6 %, seguido de un error de calibración. Intenté sustituir el módulo GPS, el módulo IMU y el cable flexible del GPS, pero no funcionó. La aeronave muestra los errores 30055 (IMU no calibrada. Calibrar IMU) y 30056 (Calibración IMU incompleta. Recalibrar IMU). Creo que se trata de un error de software y que algún fallo del software me impide solucionarlo. ¿Podría la comunidad de Github investigar el problema y ayudarme? Además, haciendo una búsqueda exhaustiva en Google, encontré una captura de pantalla de un kit de software que sugiere calibrar la IMU mediante algún software. Adjunto la captura de pantalla de los errores y el posible software que podría permitir la calibración de la IMU. Si alguien ha tenido errores similares en el pasado o sabe algo sobre el software, por favor, ayúdenme. Muchas gracias.
> 
> ![Screenshot_20220915-021910](https://user-images.githubusercontent.com/62441729/191871883-8e959946-32a6-4eb2-8922-2c4b900dfc98.jpg)
> 
> ![58301472-3b79bb00-7def-11e9-945c-091f854a1f51](https://user-images.githubusercontent.com/62441729/191871767-46e370fc-91e6-425f-974b-53067fcfd857.png)

Amigo si pudo encontrar la solución a este problema? Yo también tengo el mismo problema y pareciera que fueran motores o esc ... Pero no estoy seguro are más pruebas pero si usted encontró dónde se origina el error me ayuda porfavor

**Comment by Abiut10** [comment]
Ninguna solucion aun ?

**Comment by marianosamy96-hub** [comment]
Hai risolto? Se oreminsu imu cosa ti dice ? Hai un link di quel programma cosi vedo?
